src/System.Runtime.Extensions/src/System/IO/StreamWriter.cs

```diff
@@ -555,6 +571,11 @@ public override void WriteLine(string value)
             }
         }
 
+        public override void WriteLine(ReadOnlySpan<char> source)
+        {
+            WriteLine(new string(source.ToArray()));
```

Comment:
This is very inefficient, allocating both a `char[]` and a `string`. At the very least a better implementation would be:
```C#
Write(source);
WriteLine();
```
but even better would be like the one that takes a `string`, except using the `ReadOnlySpan<char>` as the source, and unless there's a measurable perf regression we can't overcome, both this and the string-based overload should just delegate to the same private `ReadOnlySpan<char>`-based helper that has that logic.  This one would then also do a check to make sure it's a concrete `StreamWriter`, and if it's not, just delegate to the base, e.g.
```C#
if (GetType() == typeof(StreamWriter))
{
    WriteLineCore(source);
}
else
{
    base.WriteLine(source);
}
```

Reply:
Thanks @stephentoub . Actually I had changed ```WriteLine(string)``` to use ```ReadOnlySpan``` and then make existing `string` based overload delegate to it... But after we discussed about `TextReader/Writer` to use rented array, I thought that is something to be done across the board!

Reply:
> I thought that is something to be done across the board!

TextReader/Writer are abstract base classes: these methods needed to delegate there to the existing virtuals in order for them to get the correct behavior from existing overrides.  But StreamReader/Writer are derived classes with their own implementations; as long as the instance is a concrete StreamReader/Writer and not something derived from it, we want them to have the most efficient implementations possible (and for the case where it's a derived type and thus where we still need to call an existing virtual method that may have been overridden), we just delegate to the base methods on TextReader/Writer that already do the appropriate delegation.